common/datastore/validation.go

```diff
@@ -5,8 +5,11 @@ import (
 	"gopkg.in/go-playground/validator.v9"
 )
 
-func (datastore *DataStore) Validate() error {
+func (datastore *DataStore) Validate(params ...int) error {
 	validate := validator.New()
+	if len(params) == 1{
+		return nil
+	}
```

Comment:
Could you explain what this change does?

Edit: ah I see you made this flag to skip the validation. I think we can avoid having to do this since we might need to change our strategy for this PR. See my other comment below